### PR TITLE
Fixed regression with dns server start condition

### LINKED ISSUE #3135: can't disable DNS server in consul v0.8.4 with DNS port < 0
### `consul version` for both Client and Server
Client: none
Server: v0.8.4

Server:
```
[Server `consul info` here]
agent:
        check_monitors = 0
        check_ttls = 0
        checks = 0
        services = 1
build:
        prerelease =
        revision = f436077
        version = 0.8.4
consul:
        bootstrap = false
        known_datacenters = 1
        leader = false
        leader_addr = 192.168.132.45:8300
        server = true
raft:
        applied_index = 21
        commit_index = 21
        fsm_pending = 0
        last_contact = 23.072296ms
        last_log_index = 21
        last_log_term = 117
        last_snapshot_index = 0
        last_snapshot_term = 0
        latest_configuration = [{Suffrage:Voter ID:6aeef61c-7772-83d8-977c-a7028810f8b5 Address:192.168.132.45:8300} {Suffrage:Voter ID:f77dd935-e0ec-2275-1f0e-ec8c12ed55a0 Address:192.168.132.43:8300} {Suffrage:Nonvoter ID:276f2c06-0c9d-f56e-d623-65ab0571eb54 Address:192.168.132.44:8300}]
        latest_configuration_index = 5
        num_peers = 1
        protocol_version = 3
        protocol_version_max = 3
        protocol_version_min = 0
        snapshot_version_max = 1
        snapshot_version_min = 0
        state = Follower
        term = 117
runtime:
        arch = amd64
        cpu_count = 2
        goroutines = 85
        max_procs = 2
        os = linux
        version = go1.8.3
serf_lan:
        coordinate_resets = 0
        encrypted = false
        event_queue = 0
        event_time = 2
        failed = 0
        health_score = 0
        intent_queue = 0
        left = 0
        member_time = 6
        members = 3
        query_queue = 0
        query_time = 1
serf_wan:
        coordinate_resets = 0
        encrypted = false
        event_queue = 0
        event_time = 1
        failed = 0
        health_score = 0
        intent_queue = 0
        left = 0
        member_time = 6
        members = 3
        query_queue = 0
        query_time = 1

### Operating system and Environment details
linux 64 bit contos 7

### Description of the Issue (and unexpected/desired result)
Can't seem to disable DNS server in consul v0.8.4

### Reproduction steps
create a consul.conf file with the following snippet:

        "ports":        {
                "dns":  -1,
                "https":        -1,
                "serf_lan":     8301,
                "serf_wan":     8302,
                "server":       8300
        }

start a server agent and note the following logs:

### Log Fragments
==> WARNING: Expect Mode enabled, expecting 2 servers
==> Starting Consul agent...
==> Error starting agent: agent: timeout starting DNS servers

(log_level = TRACE)

this used to work in consul v0.8.1, and setting the dns port to 0 works fine.

### PR DESCRIPTION
**Pull request by preetapan**
Found by comparing sha hashes 10540f8d5f671 and b6c69ebf5db

Fixes #3135

### CODE REVIEW

File: command/agent/agent.go [hunk 1]
```diff
@@ -291,8 +291,10 @@ func (a *Agent) Start() error {
 	}
 
 	// start DNS servers
-	if err := a.listenAndServeDNS(); err != nil {
-		return err
+	if c.Ports.DNS > 0 {
```
Comment (magiconair) [inline command/agent/agent.go]: A better way to fix this is for `config.DNSAddrs()` to return an empty array.

Comment (magiconair) [inline command/agent/agent.go]: Which it does for `Ports.DNS == 0` but not `<0`

Comment (magiconair) [inline command/agent/agent.go]: https://github.com/hashicorp/consul/blob/master/command/agent/config.go#L793-L795

File: command/agent/config.go [hunk 2]
```diff
@@ -790,7 +790,7 @@ func (p ProtoAddr) String() string {
 }
 
 func (c *Config) DNSAddrs() ([]ProtoAddr, error) {
-	if c.Ports.DNS == 0 {
+	if c.Ports.DNS < 0 {
```
Comment (magiconair) [inline command/agent/config.go]: Also, previously the code checked for `> 0` to enable. Inverting it is `<= 0`

File: command/agent/config.go [hunk 3]
```diff
@@ -790,7 +790,7 @@ func (p ProtoAddr) String() string {
 }
 
 func (c *Config) DNSAddrs() ([]ProtoAddr, error) {
-	if c.Ports.DNS == 0 {
+	if c.Ports.DNS <= 0 {
```
Comment (preetapan) [inline command/agent/config.go]: Fixed now

**Comment by magiconair** [comment]
LGTM. Can you squash merge this since only the last commit is relevant?

**Comment by slackpad** [comment]
This looked good so I went ahead and merged + updated the changelog.

**Pull request merged**